### PR TITLE
Slirp 64bit fix

### PR DESCRIPTION
**Pull request by SegHaxx**
slirp networking crashes on startup when enabled on aarch64. This is an update of #55 which seems to have gone stale, and fixes the crash. I've removed the unrelated changes from their repo, but they still show in the history, not much I can do about that.

With this patch and #227 I am able to start up System 7.6.1 and use Netscape Communicator on my Raspi4. Also tested and working on Linux x86_64.

### CODE REVIEW

File: SheepShaver/src/include/main.h [hunk 1]
```diff
@@ -41,6 +41,8 @@ struct M68kRegisters {
 	uint32 a[8];
 };
 
+// vde switch variable
+extern char* vde_sock;
```
Comment (asvitkine) [inline SheepShaver/src/include/main.h:45]: I don't see this used anywhere. Is it needed for something?

File: BasiliskII/src/slirp/ip_input.c [hunk 2]
```diff
@@ -261,33 +273,33 @@ ip_reass(ip, fp)
...
 	/*
 	 * If there is a preceding segment, it may provide some of
 	 * our data already.  If so, drop the data from the incoming
 	 * segment.  If it provides all of our data, drop us.
 	 */
-	if (q->ipf_prev != (ipasfragp_32)fp) {
-		i = ((struct ipasfrag *)(q->ipf_prev))->ip_off +
-		  ((struct ipasfrag *)(q->ipf_prev))->ip_len - ip->ip_off;
+	if (q->ipf_prev != &fp->frag_link) {
+        struct ipasfrag *pq = q->ipf_prev;
```
Comment (asvitkine) [inline BasiliskII/src/slirp/ip_input.c:301]: Nit: Indent is off here.

File: BasiliskII/src/Unix/ether_unix.cpp [hunk 3]
```diff
@@ -273,8 +273,8 @@ bool ether_init(void)
 	}
 #endif
 	else {
-		net_if_type = NET_IF_SHEEPNET;
-		printf("selected Ethernet device type sheep_net\n");
+		net_if_type = NET_IF_VDE;
+		printf("selected Ethernet device type vde\n");
```
Comment (asvitkine) [inline BasiliskII/src/Unix/ether_unix.cpp:277]: Nit: This message is similar to line 272 but with different capitalization.

Let's maybe rephrase it to mention this is the default (e.g. add " (default)" at the end).

Also, can you add a comment about why we default to VDE? I don't actually know what makes it a better default than sheep_net.

Also, how does it work if HAVE_LIBVDEPLUG is not true?

.. Or was this debugging code that should be removed?

File: BasiliskII/src/Unix/ether_unix.cpp [hunk 4]
```diff
@@ -273,8 +273,8 @@ bool ether_init(void)
 	}
 #endif
 	else {
-		net_if_type = NET_IF_SHEEPNET;
```
Comment (asvitkine) [inline BasiliskII/src/Unix/ether_unix.cpp:276]: Does it mean that NET_IF_SHEEPNET is not longer possible to select? Since there's no command line flag for it anymore?

**Comment by SegHaxx** [comment]
Yeah not sure about the VDE stuff. It looks like it may be unfinished. That part should perhaps be reversed yes. I was just happy to see slirp working. :)

Also if you do ':set list' in vim you find that the indentation in the slirp sources is already very random. I tried to clean it up best i could in the patch but these files really need to be entirely reformatted. Which would be best done in it's own patch.
